includes/settings.php

```diff
@@ -21,82 +21,108 @@ public function __construct() {
-			array( &$this, 'display_settings_page' )			// callback
+			__( 'Mad Mimi Settings', 'mimi' ),        // <title> tag
+			__( 'Mad Mimi Settings', 'mimi' ),        // menu label
+			'manage_options',                         // required cap to view this page
+			$this->slug = 'mad-mimi-settings',        // page slug
+			array( &$this, 'display_settings_page' )  // callback
 		);
 
-		add_action( "load-$this->hook", array( $this, 'page_load' ) );
+		add_action( 'load-' . $this->hook, array( $this, 'page_load' ) );
+
 	}
 
 	public function page_load() {
+
 		// main switch for some various maintenance processes
 		if ( isset( $_GET['action'] ) ) {
+
 			$settings = get_option( $this->slug );
 
 			switch ( $_GET['action'] ) {
-				case 'debug-reset':
-					if ( ! $this->mimi->debug )
+
+				case 'debug-reset' :
+
+					if ( ! $this->mimi->debug ) {
 						return;
+					}
 
 					if ( isset( $settings['username'] ) ) {
-						delete_transient( "mimi-{$settings['username']}-lists" );
+						delete_transient( 'mimi-' . $settings['username'] . '-lists' );
 					}
+
 					delete_option( $this->slug );
 
 					break;
-				case 'debug-reset-transients':
-					if ( ! $this->mimi->debug )
+
+				case 'debug-reset-transients' :
+
+					if ( ! $this->mimi->debug ) {
 						return;
+					}
 
 					if ( isset( $settings['username'] ) ) {
+
 						// remove all lists
-						delete_transient( "mimi-{$settings['username']}-lists" );
+						delete_transient( 'mimi-' . $settings['username'] . '-lists' );
 
 						// mass-removal of all forms
 						foreach ( Mad_Mimi_Dispatcher::get_forms()->signups as $form ) {
-							delete_transient( "mimi-form-{$form->id}" );
+							delete_transient( 'mimi-form-' . $form->id );
 						}
 
 						add_settings_error( $this->slug, 'mimi-reset', __( 'All transients were removed.', 'mimi' ), 'updated' );
 					}
+
 					break;
 
-				case 'refresh':
+				case 'refresh' :
+
 					// remove only the lists for the current user
 					if ( isset( $settings['username'] ) ) {
-						if ( delete_transient( "mimi-{$settings['username']}-lists" ) )
+
+						if ( delete_transient( 'mimi-' . $settings['username'] . '-lists' ) ) {
 							add_settings_error( $this->slug, 'mimi-reset', __( 'Forms list was successfully updated.', 'mimi' ), 'updated' );
+						}
+
 					}
 
-					foreach ( Mad_Mimi_Dispatcher::get_forms()->signups as $form ) {
-						delete_transient( "mimi-form-{$form->id}" );
+					foreach ( (array) Mad_Mimi_Dispatcher::get_forms()->signups as $form ) {
+						delete_transient( 'mimi-form-' . $form->id );
 					}
 
 					break;
 
-				case 'edit_form':
-					if ( ! isset( $_GET['form_id'] ) )
+				case 'edit_form' :
+
+					if ( ! isset( $_GET['form_id'] ) ) {
 						return;
+					}
 
 					$tokenized_url = add_query_arg( 'redirect', sprintf( '/signups/%d/edit', absint( $_GET['form_id'] ) ), Mad_Mimi_Dispatcher::user_sign_in() );
 
+					// Not wp_safe_redirect as it's an external site
 					wp_redirect( $tokenized_url );
```

Comment:
@joehoyle it's redirecting to an external site, I guess we'd need to add that domain to the safe redirect list?